org.eclipse.swt.gtk.linux.x86/src/org/eclipse/swt/internal/Lock.d

```diff
@@ -51,7 +46,7 @@ public int lock() {
             while (count > 0) {
                 try {
                     cond.wait();
-                } catch (SyncException e) {
+                } catch (SyncError e) {
```

Comment:
Anything inheriting from `Error` should not be caught (see above comment). I don't know the best way to deal with this. On Posix, it looks like this can only fail if the condition or the mutex passed to the underlying `pthread_cond_wait` function are invalid. On Windows it looks like it will never throw `SyncError` (although I think some underlying OS specific exception can be thrown, but that will crash the program).

Reply:
I was just looking at the source code to see when it threw :)

As you said above, since an `Error` isn't supposed to be caught then wouldn't it be best to not catch them?

The old code has `wait()` throw a `SyncException` in the same place that throws a `SyncError` in the Condition class (i.e. [Line 234]).

[Line 234]: https://github.com/dlang/druntime/blob/d4487d8931a42547f296773ecb8d576cda8a2f24/src/core/sync/condition.d#L234

Reply:
> As you said above, since an Error isn't supposed to be caught then wouldn't it be best to not catch them?

That will be a different behavior compared to the current code. But it might be best anyway. I'll have to think about this a bit.

Reply:
>  I'll have to think about this a bit.

No problem.

Reply:
Let's not catch `SyncError`. I think this can only happen if there's a bug in the implementation of `wait`.

Reply:
Okay. I've removed the `try/catch` statements.